site/src/pages/ExternalAuthPage/ExternalAuthPage.tsx

```diff
@@ -72,6 +76,34 @@ const ExternalAuthPage: FC = () => {
     !getExternalAuthProviderQuery.data.authenticated &&
     !getExternalAuthProviderQuery.data.device
   ) {
+    if (window.location.search.includes("redirected=true")) {
```

Comment:
We can check the query directly. I believe there's a `useSearch` that does this.

Reply:
Done